File: backend/src/main/java/ch/rodano/core/services/bll/configurator/RuleDefinitionPropertyService.java
package ch.rodano.core.services.bll.configurator;

import java.util.List;
import java.util.UUID;

import ch.rodano.api.config.RuleDefinitionPropertyDTO;

public interface RuleDefinitionPropertyService {

	List<RuleDefinitionPropertyDTO> getRuleDefinitionProperties(UUID projectId);

	RuleDefinitionPropertyDTO getRuleDefinitionProperty(UUID projectId, UUID ruleDefinitionPropertyId);

	RuleDefinitionPropertyDTO createRuleDefinitionProperty(UUID projectId, RuleDefinitionPropertyDTO dto);

	RuleDefinitionPropertyDTO updateRuleDefinitionProperty(UUID projectId, UUID ruleDefinitionPropertyId, RuleDefinitionPropertyDTO dto);

	void deleteRuleDefinitionProperty(UUID projectId, UUID ruleDefinitionPropertyId);
}

File: frontends/main/src/app/configurator/services/api/rule-definition-property.service.ts
import {Injectable} from '@angular/core';
import {HttpClient} from '@angular/common/http';
import {Observable} from 'rxjs';
import {RuleDefinitionProperty} from '@core/model/rule-definition-property';

@Injectable({
	providedIn: 'root'
})
export class RuleDefinitionPropertyService {
	constructor(private http: HttpClient) {}

	getRuleDefinitionProperties(projectId: string): Observable<RuleDefinitionProperty[]> {
		return this.http.get<RuleDefinitionProperty[]>(`/api/superuser/configurator/projects/${projectId}/config/rule-def-properties`);
	}

	getRuleDefinitionProperty(projectId: string, ruleDefinitionPropertyId: string): Observable<RuleDefinitionProperty> {
		return this.http.get<RuleDefinitionProperty>(`/api/superuser/configurator/projects/${projectId}/config/rule-def-properties/${ruleDefinitionPropertyId}`);
	}

	createRuleDefinitionProperty(projectId: string, ruleDefinitionProperty: RuleDefinitionProperty): Observable<RuleDefinitionProperty> {
		return this.http.post<RuleDefinitionProperty>(`/api/superuser/configurator/projects/${projectId}/config/rule-def-properties`, ruleDefinitionProperty);
	}

	updateRuleDefinitionProperty(projectId: string, ruleDefinitionPropertyId: string, ruleDefinitionProperty: RuleDefinitionProperty): Observable<RuleDefinitionProperty> {
		return this.http.put<RuleDefinitionProperty>(`/api/superuser/configurator/projects/${projectId}/config/rule-def-properties/${ruleDefinitionPropertyId}`, ruleDefinitionProperty);
	}

	deleteRuleDefinitionProperty(projectId: string, ruleDefinitionPropertyId: string): Observable<void> {
		return this.http.delete<void>(`/api/superuser/configurator/projects/${projectId}/config/rule-def-properties/${ruleDefinitionPropertyId}`);
	}
}
